cpuinfo.go

```diff
@@ -33,9 +34,15 @@ func cpuVariant() string {
 	cpuVariantOnce.Do(func() {
 		if isArmArch(runtime.GOARCH) {
 			var err error
-			cpuVariantValue, err = getCPUVariant()
+			cpuVariantValue, err = getArmCPUVariant()
 			if err != nil {
-				log.L.Errorf("Error getCPUVariant for OS %s: %v", runtime.GOOS, err)
+				log.L.Errorf("Error getArmCPUVariant for OS %s: %v", runtime.GOOS, err)
+			}
+		} else if isAmd64Arch(runtime.GOARCH) {
+			var err error
+			cpuVariantValue, err = getAmd64MicroArchLevel()
```

Comment:
This should not change the value of `DefaultSpec()`. Default variant for amd64 is still `v1` and `v8` for arm64 even if chip is `v9` compatible. Eg. image created on amd64 system will always be `linux/amd64/v1` (normalized to `linux/amd64`) if user didn't specify that they want to create specific variant (or multiple images with different variants). This does not change depending on whether the building system happens to have AVX support or not.

I'd recommend creating a new function like `MaximumSpec` similar to https://github.com/moby/moby/blob/v26.1.4/distribution/pull_v2.go#L1089-L1096 that can be used in the cases where comparison to the maximum compatible version is needed (eg. when pulling multi-arch image manifest).